packages/flutter/lib/src/widgets/magnifier.dart

```diff
@@ -0,0 +1,545 @@
+  final MagnifierDecoration decoration;
+
+  /// The size of the magnifier.
+  ///
+  /// This does not include added border; it only includes
+  /// the size of the magnifier.
+  final Size size;
+
+  /// The offset of the magnifier from [RawMagnifier]'s center.
+  ///
+  /// If [focalPoint] is [Offset.zero], then the [focalPoint]
+  /// will point directly below this [RawMagnifier].
+  final Offset focalPoint;
+
+  /// An optional widget to posiiton inside the len of the [RawMagnifier].
+  ///
+  /// This is positioned over the [RawMagnifier] - it may be useful for tinting the
+  /// [RawMagnifier], or drawing a crosshair like UI.
+  final Widget? child;
+
+  /// How "zoomed in" the magnification subject is in the lens.
+  final double magnificationScale;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      clipBehavior: Clip.none,
+      alignment: Alignment.center,
+      children: <Widget>[
+          Opacity(
+            opacity: decoration.opacity,
+            child: _Magnifier(
+              shape: decoration.shape,
+              focalPoint: focalPoint,
+              magnificationScale: magnificationScale,
+              child: SizedBox.fromSize(
+                size: size,
+                child: child,
+              ),
+            ),
+        ),
+
+      // Because `BackdropFilter` will filter any widgets before it, we should
+      // apply the style after (i.e. in a younger sibling) to avoid the magnifier
+      // from seeing its own styling.
+        Opacity(
+          opacity: decoration.opacity,
+          child: _MagnifierStyle(
+              decoration,
+              size: size,
+          ),
+        )
+      ],
+    );
+  }
+}
+
+class _MagnifierStyle extends StatelessWidget {
+  const _MagnifierStyle(this.decoration, {required this.size});
+
+  final MagnifierDecoration decoration;
+  final Size size;
+
+  @override
+  Widget build(BuildContext context) {
+    double largestShadow = 0;
+    for (final BoxShadow shadow in decoration.shadows ?? <BoxShadow>[]) {
+      largestShadow = math.max(
+          largestShadow,
+          (shadow.blurRadius + shadow.spreadRadius) +
+              math.max(shadow.offset.dy.abs(), shadow.offset.dx.abs()));
+    }
+
+    return ClipPath(
+      clipBehavior: Clip.hardEdge,
+      clipper: _DonutClip(
+        shape: decoration.shape,
+        spreadRadius: largestShadow,
+      ),
+      child: DecoratedBox(
+        decoration: decoration,
+        child: SizedBox.fromSize(
+          size: size,
+        ),
+      ),
+    );
+  }
+}
+
+/// A `clipPath` that looks like a donut if you were to fill its area.
+///
+/// This is necessary because the shadow must be added after the magnifier is drawn,
+/// so that the shadow does not end up in the magnifier. Without this clip, the magnifier would be
+/// entirely covered by the shadow.
+///
+/// The negative space of the donut is clipped out (the donut hole, outside the donut).
```

Comment:
I couldn't find a way to make the clip unbounded. I would like to make it so that we don't need to clip the outside, but I wasn't able to find a good solution,